low-level/sensors/impl/generic/node_hba.py

```diff
@@ -0,0 +1,441 @@
+from framework.utils.store_factory import file_store
+from framework.utils.tool_factory import ToolFactory
+from framework.utils.os_utils import OSUtils
+from framework.utils.mon_utils import MonUtils
+from message_handlers.node_data_msg_handler import NodeDataMsgHandler
+from framework.messaging.egress_processor import EgressProcessor
+from framework.platforms.server.component_factory import ServerCompFactory
+from json_msgs.messages.sensors.hba_data import HBADataMsg
+from sensors.Ihba import IHBASensor
+
+# Override default store
+store = file_store
+
+
+@implementer(IHBASensor)
+class HBASensor(SensorThread, InternalMsgQ):
+    """
+    HBA sensor which runs on its own thread periodically and
+    is responsible for sensing events like HBA card insertion,
+    removable, host port running and not running.
+    """
+
+    SENSOR_NAME = "HBASensor"
+    PRIORITY = 1
+
+    HBA_RESOURCE_TYPE = "node:hw:hba"
+    HBA_PORT_RESOURCE_TYPE = "node:hw:hba:port"
+
+    # FRU alert types
+    FRU_MISSING = "missing"
+    FRU_INSERTION = "insertion"
+    FRU_FAULT = "fault"
+    FRU_FAULT_RESOLVED = "fault_resolved"
+
+    FRU_ALERTS = [FRU_MISSING, FRU_INSERTION, FRU_FAULT, FRU_FAULT_RESOLVED]
+
+    # Section in the configuration store
+    SYSTEM_INFORMATION = "SYSTEM_INFORMATION"
+    POLLING_INTERVAL = "polling_interval"
+
+    CACHE_DIR_NAME  = "server"
+    DEFAULT_POLLING_INTERVAL = '20'
+    PROBE = "probe"
+
+    # Dependency list
+    DEPENDENCIES = {
+            "plugins": ["NodeControllerMsgHandler", "EgressProcessor"],
+            "rpms": []
+        }
+
+
+    @staticmethod
+    def name():
+        """@return: name of the module."""
+        return HBASensor.SENSOR_NAME
+
+
+    @staticmethod
+    def impact():
+        """Returns impact of the module."""
+        return "Server HBA card and initiators can not be monitored."
+
+
+    def __init__(self, utility_instance=None):
+        """Initialize HBA sensor instance."""
+        super(HBASensor, self).__init__(self.SENSOR_NAME, self.PRIORITY)
+        self._utility_instance = utility_instance
+        self._hba = None
+        self.hosts = []
+        self.host_data = []
+        self.hba_data_msg = None
+
+        # Flag to indicate suspension of module
+        self._suspended = False
+
+
+    def initialize(self, conf_reader, msgQlist, product):
+        """initialize configuration reader and internal msg queues"""
+
+        # Initialize ScheduledMonitorThread and InternalMsgQ
+        super(HBASensor, self).initialize(conf_reader)
+        super(HBASensor, self).initialize_msgQ(msgQlist)
+
+        # Get the hba implementor from configuration
+        hba_utility = Conf.get(SSPL_CONF,
+                               f"{self.name().upper()}>{self.PROBE}", "sysfs")
+
+        self.polling_interval = int(Conf.get(SSPL_CONF,
+            f"{self.SENSOR_NAME.upper()}>{self.POLLING_INTERVAL}",
+            self.DEFAULT_POLLING_INTERVAL))
+
```

Comment:
alignment at 116 and 119 is different for brackets second lines.

Reply:
Done.